tools/mercury/README.md

```diff
@@ -0,0 +1,520 @@
+# Mercury
+##interactive Golang Testing System for Qpid Dispatch Router
+
+
+
+<br/>
+
+
+###Requirements
+
+1. the go language on your machine
+2. installed proton and dispatch router software
+
+
+
+<br/>
+
+###Mercury Audience
+
+I see Mercury as a tool for developers. The user has one or more installed versions of the Dispatch Router + Proton code and wantsto easily set up a complex network including Dispatch routers, edge routers, and clients with nontrivial addressing patterns. Currently I think that creation of nontrivial networks is difficult enough that it discourages extensive testing during development.
+
+Especially I would like to see a more interactive form of development testing, in which a developer easily creates a network, and easily iterates through a cycle of altering the network, running it, and seeing results from the run. At the end of such a 'session' all of the developer's actions have been captured and will be reproducible later if desired. The captured session is a runnable script, and can be edited and used as a standardized test.
+
+
+<br/>
+
+
+
+###The C Client
+
+The client that I use is written in C to the proactor interface and has been heavily adapted from an original by Alan Conway. It needs to be built before Mercury will be able to do anything useful. Look at the directory mercury/clients, look at the file in there called "m", adapt it for your system, and run it so you get an executable.
+
+Having my own client allows me to do things like:
+
+1. throttle send-speed with the "--throttle" argument
+2. Tell the client to form multiple links with multiple "--address" arguments.
+3. Tell it where to send its log files and so on.
+
+
+<br/>
+
+
+###Starting Mercury
+
+The directory from which to run Mercury is also called mercury.  So it's mercury/mercury.  And the run-script that I use is 'r'.
```

Comment:
run script is r ? Where is r ?

Reply:
I changed this to say "example_run_script" instead of "r", and then checked that in to the code tree.  And that run script calls "example_script" which is also checked in.